### PR TITLE
docs: Fr/editing cookbook

### PR DESCRIPTION
**Pull request by rodriguesfred**
## Description
Adding further examples to the plotly section of the cookbook.
Added a focus palette although I'm not sure it makes much sense. Probably best to just use main and call the colors.

Note that small multiples has an issue were I can't get axis on both rows.

### Related Issues

- Fixes #
- Closes #
- Relates to #63 

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement
- [ ] 🔨 Build/CI changes
- [ ] 🎨 Style/formatting changes

## Changes Made

- 
- 
- 

### Test Coverage
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing performed
- [ ] No tests needed (explain why below)

### Testing Details

**Test Cases:**
1. 
2. 
3. 

**Manual Testing Results:**

## Breaking Changes

- [x] No breaking changes
- [ ] Breaking changes (describe below)

**Breaking Change Details:**

## Documentation
- [x] Documentation updated
- [ ] README updated
- [ ] No documentation changes needed

**Documentation Changes:**

## Dependencies
- [x] No new dependencies
- [ ] New dependencies added (list below)
- [ ] Dependencies updated (list below)
- [ ] Dependencies removed (list below)

**Dependency Changes:**

### Code Quality
- [x] Code follows the project's [style guidelines](https://3.basecamp.com/4322319/buckets/25844562/documents/8777425747)
- [ ] Self-review of code completed
- [x] Code is properly commented
- [x] No unnecessary console.log/print statements
- [ ] No commented-out code (unless intentionally documented)

### Testing & Quality Assurance
- [ ] Tests pass locally
- [ ] Tests added for new functionality
- [ ] Edge cases considered and tested
- [ ] Error handling implemented where appropriate

### CODE REVIEW

File: src/afcharts/cookbook/03-plotly-usage.qmd [hunk 1]
```diff
@@ -188,3 +344,221 @@ fig.show()
...
+```{python}
+#| echo: false
+
+fig.show()
+
+```
+
+<div class="chart-source">Source: Gapminder</div>
+
+This pie chart uses the afcharts theme, showing the proportions of European countries with a life expectancy under and over 75. The segment colours are from the Analysis Function categorical2 palette, with the smaller under 75 segment in dark blue, and the larger over 75 segment in orange. This is indicated by a legend to the right of the pie chart. There is whitespace separating the segments from each other.
```
Comment (nickhenden) [inline src/afcharts/cookbook/03-plotly-usage.qmd:500]:
```suggestion
This pie chart uses the afcharts theme, showing the proportions of European countries with a life expectancy under and over 75. The segment colours are from the Analysis Function categorical palette, with the smaller under 75 segment in dark blue, and the larger over 75 segment in orange. This is indicated by a legend to the right of the pie chart. There is whitespace separating the segments from each other.
```

File: src/afcharts/cookbook/03-plotly-usage.qmd [hunk 2]
```diff
@@ -188,3 +344,221 @@ fig.show()
...
+# Filter out Oceania and aggregate population by continent and year
+df_filtered = df[df['continent'] != 'Oceania']
+df_grouped = df_filtered.groupby(['continent', 'year'], observed=True)['pop'].sum().reset_index()
+
+# Define the continents to plot
+continents = df_grouped['continent'].unique()
+
+
+# Create a 2x2 subplot layout
+fig = make_subplots(rows=2, cols=2, subplot_titles=continents, shared_yaxes=True)
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply src/afcharts/cookbook/03-plotly-usage.qmd:378]: I say we can add it to issues.

[the remark this answers is not in the document]
Comment (nickhenden) [reply src/afcharts/cookbook/03-plotly-usage.qmd:380]: Yeah fair enough, please add it to the issues log

**Review by nickhenden — APPROVED** [review]
Approved. Add an issue for the titles getting cut off and then resolve the last conversation before you merge.
